### PR TITLE
new hbase architecture support

### PR DESCRIPTION
**Pull request by hthayko**

### CODE REVIEW

**Comment by hthayko** [comment]
Hbase Revolution Brough to you by:
https://github.com/ripple/crawl-reporter/pull/9
https://github.com/ripple/crawler-api/pull/9
https://github.com/ripple/rippled-network-crawler/pull/54

File: src/lib/process_crawl.js [hunk 1]
```diff
@@ -221,9 +237,10 @@ function getMetrics(crawl_row) {
 }
 
 module.exports = function(crawl_row, log) {
+  if(!crawl_row) return undefined;
```
Comment (wilsonianb) [inline src/lib/process_crawl.js]: Why not throw an error here? Is this a case that is expected to happen?

Comment (hthayko) [inline src/lib/process_crawl.js]: yeah, that happens in the beginning of the times, when we get the first crawl and there is no a previous one for it, so process_crawl() is called for undefined and returns undefined

File: src/live.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
 var moment = require('moment');
 var Promise = require('bluebird');
 var graphite = require('graphite');
+var HbaseClient = require('crawler-hbase').Client;
 
 module.exports = function(max, queueUrl, dbUrl, graphiteUrl, log) {
   var graphiteClient = graphite.createClient(graphiteUrl);
-  var logsql = false;
-
+  var hbaseClient = new HbaseClient(dbUrl);
```
Comment (wilsonianb) [inline src/live.js]: Is this outside `processMessage` so that we don't establish a new connection for each message/crawl?

Comment (hthayko) [inline src/live.js]: it's outside for not calling constructor for each message/crawl. 
HbaseClient is designed to share one connection pool amongst all the instances.

File: src/lib/report_graphite.js [hunk 3]
```diff
@@ -4,25 +4,24 @@ var _ = require('lodash');
...
     };
   });
-
   graphiteClient.write(metrics, moment(crawlMetrics.crawl.start).valueOf(), function(error) {
     if (error) {
       throw new Error(error);
     } else {
-      console.log('Reported  crawl %d \t at %s \t to graphite',
+      console.log('Reported  crawl %s \t at %s \t to graphite',
         crawlMetrics.crawl.id, moment().format());
```
Comment (wilsonianb) [inline src/lib/report_graphite.js]: `crawlMetrics.crawl.id` -> `crawlMetrics.crawl.rowkey`?

Comment (hthayko) [inline src/lib/report_graphite.js]: crawlMetrics.crawl is in following format:
{
    'id': crawl_row.rowkey,
    'start': hbaseUtils.keyToStart(crawl_row.rowkey),
    'end': hbaseUtils.keyToEnd(crawl_row.rowkey),
    'entry': crawl_row.entry_ipp
  };

Comment (hthayko) [inline src/lib/report_graphite.js]: id and rowkey are used interchangeably. I guess that's bad ?

**Comment by wilsonianb** [comment]
What do you think about moving `buildChangedNodes` and `buildNodeStats` from `storeProcessedCrawl` into the reporter and just calling the crawler-hbase for the store functions?

**Comment by wilsonianb** [comment]
LGTM :+1:

**Pull request merged**